spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -318,16 +353,31 @@ public final void addQueues(Queue... queues) {
 	public boolean removeQueueNames(String... queueNames) {
 		Assert.notNull(queueNames, "'queueNames' cannot be null");
 		Assert.noNullElements(queueNames, "'queueNames' cannot contain null elements");
-		return this.queueNames.removeAll(Arrays.asList(queueNames));
+		AtomicBoolean result = new AtomicBoolean();
+		if (this.queues.size() > 0) {
+			Set<String> toRemove = new HashSet<>(Arrays.asList(queueNames));
+			this.queues.stream().forEach(q -> {
+				if (toRemove.contains(q.getName()) || toRemove.contains(q.getDeclaredName())) {
+					this.queues.remove(q);
+					result.set(true);
+				}
+			});
+		}
+		return result.get();
 	}
 
 	/**
 	 * Remove queue(s) from this container's list of queues.
 	 * @param queues The queue(s) to remove.
 	 * @return the boolean result of removal on the target {@code queueNames} List.
 	 */
-	public final boolean removeQueues(Queue... queues) {
-		return removeQueueNames(collectQueueNames(queues));
+	public boolean removeQueues(Queue... queues) {
+		Assert.notNull(queues, "'queues' cannot be null");
+		Assert.noNullElements(queues, "'queues' cannot contain null elements");
+		return removeQueueNames(Arrays.stream(queues)
+				.map(q -> q.getName())
```

Comment:
Well, that's confusing. In other places you also consult a `delcaredName` property. Why not here?
More over we have just realized that for broker-based names we always are going to have the `name` as an empty string.

Would you mind to elaborate more what's going on here?

Something similar I see in the `DMLC` - we remove using the `name` and don't check a `declaredName`